test/trento/application/usecases/clusters_test.exs

```diff
@@ -30,6 +33,47 @@ defmodule Trento.ClustersTest do
     end
   end
 
+  describe "checks execution with wanda adapter" do
+    test "should start a checks execution on demand" do
+      %{id: cluster_id, provider: provider, selected_checks: selected_checks} = insert(:cluster)
+      [%{id: host_id_1}, %{id: host_id_2}] = insert_list(2, :host, cluster_id: cluster_id)
+
+      expect(Trento.Integration.Checks.Mock, :request_execution, fn _,
+                                                                    expected_cluster_id,
+                                                                    expected_provider,
+                                                                    expected_hosts,
+                                                                    expected_checks ->
+        assert ^cluster_id = expected_cluster_id
+        assert ^provider = expected_provider
+        assert [%{host_id: host_id_1}, %{host_id: host_id_2}] == expected_hosts
+        assert ^selected_checks = expected_checks
+        :ok
+      end)
+
+      assert :ok = Clusters.Wanda.request_checks_execution(cluster_id)
+    end
+
+    test "should not start checks execution if the cluster is not registered" do
+      assert {:error, :cluster_not_found} = Clusters.Wanda.request_checks_execution(UUID.uuid4())
+    end
+
+    test "should not start checks execution if no checks are selected" do
+      %{id: cluster_id} = insert(:cluster, selected_checks: [])
+
+      assert :ok = Clusters.Wanda.request_checks_execution(cluster_id)
```

Comment:
Same as before. To me what would tell that `should not start checks execution` is the fact that we do not call `Trento.Integration.Checks.request_execution`, next to the `:ok = Clusters.Wanda...`